frodo.js

```diff
@@ -1,95 +1,103 @@
-const shuffledArray = array.sort(() => Math.random() - 0.5);
```

Comment:
SCRIPT-Consider extracting this repeated logic into a separate function to avoid duplication.